inspirehep/modules/workflows/tasks/arxiv.py

```diff
@@ -250,31 +248,72 @@ def _author_list(obj, eng):
             xml_files_list = [path for path in file_list if path.endswith('.xml')]
             obj.log.info('Found xmlfiles: {0}'.format(xml_files_list))
 
+            builder = LiteratureBuilder()
             extracted_authors = []
+            author_id_paths = ["./authorIDs/authorID", "./authorids/authorid"]
+
             for xml_file in xml_files_list:
                 with open(xml_file, 'r') as xml_file_fd:
                     xml_content = xml_file_fd.read()
-
                 match = REGEXP_AUTHLIST.findall(xml_content)
                 if match:
                     obj.log.info('Found a match for author extraction')
-                    try:
-                        authors_xml = convert(xml_content, stylesheet)
-                    except XMLSyntaxError:
-                        # Probably the %auto-ignore comment exists, so we skip the
-                        # first line. See: inspirehep/inspire-next/issues/2195
-                        authors_xml = convert(
-                            xml_content.split('\n', 1)[1],
-                            stylesheet,
+
+                if isinstance(xml_content, str):
+                    xml_content = xml_content.decode('utf-8')
+
+                # Probably the %auto-ignore comment exists, so we skip the
+                # first line. See: inspirehep/inspire-next/issues/2195
+                if "%auto-ignore" in xml_content:
+                    content = Selector(text=xml_content.split('\n', 1)[1], type="xml")
+                else:
+                    content = Selector(text=xml_content, type="xml")
+
+                content.remove_namespaces()
+                # Going through all the authors in the file
+                for author in content.xpath("//Person"):
+
+                    ids = []
+                    affiliations = []
+                    # Checks which path is the correct one for this file
+                    # to get author ids
+                    if author.xpath("./authorIDs/authorID") == []:
+                        index = 1
+                    else:
+                        index = 0
+
+                    # Getting all the author ids
+                    for id in author.xpath(author_id_paths[index]):
+                        if id.xpath("./@source").get() == u'CCID':
+                            ids.append(['CERN', id.xpath("./text()").get()])
+                        elif id.xpath("./@source").get() == u'INSPIRE':
+                            ids.append([id.xpath("./@source").get() + ' ID', id.xpath("./text()").get()])
```

Comment:
can you also handle various placeholders for empty data, like in https://github.com/inspirehep/inspire-next/blob/a442fd80476e48fe37eb9a2c6e0854c487eb2caf/inspirehep/modules/workflows/utils/stylesheets/authorlist2marcxml.xsl#L71 ?

That is, if the tag contains `undefined`, `UNDEFINED`, `INSPIRE-`, `inspire-` or ``, skip it (instead of generating a validation error down the line).